spec/ysh-methods.test.sh

```diff
@@ -99,6 +99,20 @@ setvar is_a_val = "xyz"
 (Bool)   True
 ## END
 
+#### Lis->indexOf
```

Comment:
That makes total sense!
`x->transform` vs `x => pure_func`. it also reminds me a bit of the weird haskell chains (though IIRC this goes the other way).

So instead of a method this should become a function then. Let me see what I can do!

Reply:
nvm it's just syntactic sugar. I genuinly thought `=>` is for differently implemented funcs

Reply:
but that means something like `x => strip => split` would technically work but probably give an unexpected result? or is the idea to enforce one or the other at some point?

Reply:
`x => strip() => split()` should work like you expect?   what would be unexpected?